pdc_client/bin/pdc_client

```diff
@@ -0,0 +1,182 @@
+        return {}
+    try:
+        return json.loads(data)
+    except ValueError, err:
+        print "Request data is not a valid JSON input."
+        print err
+        sys.exit(1)
+
+
+if __name__ == "__main__":
+    parser = optparse.OptionParser(version=__version__)
+    parser.add_option("-s", "--server", help="PDC instance url or shortcut.")
+    parser.add_option("-x", "--request", default="GET", help="Request method.")
+    parser.add_option("-r", "--resource",
+                      help=("Resource name.\n" +
+                            "NOTE: All available resources can be got by a request " +
+                            "to the API Root. Since API Root does not belong to any " +
+                            "resources, while requesting, please omit the `-r` option.\n" +
+                            "The response of API Root request is a dict with resource names " +
+                            "as keys and resource URIs as values."),
+                      default="")
+    parser.add_option("-d", "--data", help="Request data to server.")
+    parser.add_option("-f", "--file",
+                      help="File to load request data from (or - for stdin).")
+    parser.add_option("-t", "--traceback",
+                      help="Show the traceback when an error occurs",
+                      action="store_true", default=False)
+    parser.add_option("--debug", help="Show request headers and path for "
+                                      "debugging.",
+                      action="store_true", default=False)
+    options, args = parser.parse_args()
+
+    if len(args):
+        print "Can not parse the positional arguments: %s" % ", ".join(args)
+        parser.print_help()
+        sys.exit(1)
+
+    if not options.server:
+        print "Error: -s is required.\n"
+        parser.print_help()
+        sys.exit(1)
+
+    if options.data and options.file:
+        print "Error: can not load data from file and command line at the same time."
+        sys.exit(1)
+
+    config = read_config_file(options.server)
+    if config:
+        try:
+            set_option("url", config[CONFIG_URL_KEY_NAME])
+        except KeyError:
+            print "'%s' must be specified in configuration file." % CONFIG_URL_KEY_NAME
+            sys.exit(1)
+    else:
+        set_option("url", options.server)
+    set_option("insecure", config.get(CONFIG_INSECURE_KEY_NAME))
+    set_option("develop", config.get(CONFIG_DEVELOP_KEY_NAME))
+    set_option("token", config.get(CONFIG_TOKEN_KEY_NAME))
+
+    data = load_data(options)
+
+    try:
+        client, session = make_client()
+    except BeanBagException as e:
+        print "%d %s" % (e.response.status_code, e.response.content)
+    except Exception as e:
+        print str(e)
+    else:
+        with RequestMethod(client, session, options.resource,
+                           options.traceback, options.debug) as request:
+            print json.dumps(request.dispatch(options.request, data),
```

Comment:
The task of PDC-960 will edit this script, I am waiting for this script merged to master branch.